client/state/data-layer/wpcom-http/test/http.js

```diff
@@ -0,0 +1,213 @@
+			expect( apiVersionPattern.test( '1' ) ).to.be.false;
+		} );
+
+		it( 'should accept `v1` as a valid version', () => {
+			expect( apiVersionPattern.test( 'v1' ) ).to.be.true;
+		} );
+
+		it( 'should not accept `v1.` as a valid version', () => {
+			expect( apiVersionPattern.test( 'v1.' ) ).to.be.false;
+		} );
+
+		it( 'should accept `v1.3` as a valid version', () => {
+			expect( apiVersionPattern.test( 'v1.3' ) ).to.be.true;
+		} );
+	} );
+
+	describe( 'apiNamespace shape', () => {
+		it( 'should not accept `wpcom` as a valid namespace', () => {
+			expect( apiNamespacePattern.test( 'wpcom' ) ).to.be.false;
+		} );
+
+		it( 'should accept `wpcom/v1` as a valid namespace', () => {
+			expect( apiNamespacePattern.test( 'wpcom/v1' ) ).to.be.true;
+		} );
+
+		it( 'should not accept `wpcom/` as a valid namespace', () => {
+			expect( apiNamespacePattern.test( 'wpcom/' ) ).to.be.false;
+		} );
+
+		it( 'should not accept `wpcom/1` as a valid namespace', () => {
+			expect( apiNamespacePattern.test( 'wpcom/1' ) ).to.be.false;
+		} );
+	} );
+
+	describe( 'request action', () => {
+		it( 'should return a valid redux action defining apiVersion', () => {
+			const action = http( {
+				apiVersion: 'v2',
+				method: 'GET',
+				path: '/path/to/endpoint',
+			} );
+
+			expect( action ).to.eql( {
+				type: WPCOM_HTTP_REQUEST,
+				body: {},
+				method: 'GET',
+				path: '/path/to/endpoint',
+				query: {
+					apiVersion: 'v2'
+				},
+				options: {},
+				onSuccess: null,
+				onFailure: null,
+				onProgress: null,
+			} );
+		} );
+
+		it( 'should return a valid redux action defining apiNamespace', () => {
+			const action = http( {
+				apiNamespace: 'wpcom/v2',
+				method: 'GET',
+				path: '/path/to/endpoint',
+			} );
+
+			expect( action ).to.eql( {
+				type: WPCOM_HTTP_REQUEST,
+				body: {},
+				method: 'GET',
+				path: '/path/to/endpoint',
+				query: {
+					apiNamespace: 'wpcom/v2'
+				},
+				options: {},
+				onSuccess: null,
+				onFailure: null,
+				onProgress: null,
+			} );
+		} );
+
+		it( 'should return a valid redux action passing random parameters into the query', () => {
+			const action = http( {
+				apiVersion: 'v2',
+				method: 'GET',
+				path: '/path/to/endpoint',
+				query: {
+					number: 10,
+					context: 'display',
+					fields: [ 'foo', 'bar' ],
+				}
+			} );
+
+			expect( action ).to.eql( {
+				type: WPCOM_HTTP_REQUEST,
+				body: {},
+				method: 'GET',
+				path: '/path/to/endpoint',
+				query: {
+					apiVersion: 'v2',
+					number: 10,
+					context: 'display',
+					fields: [ 'foo', 'bar' ],
+				},
+				options: {},
+				onSuccess: null,
+				onFailure: null,
+				onProgress: null,
+			} );
+		} );
+	} );
+
+	describe( 'bad request action', () => {
+		it( 'should throw Error when `apiVersion` and `apiNamespace` are not defined.', () => {
+			const boundRequest = http.bind( http, {
+				path: '/path/to/endpoint',
+				method: 'GET',
+			} );
```

Comment:
this is a bit odd to see here - as if it were working around a problem that didn't exist (simply because of the way `bind` is given `http` when in fact we don't care about that.

how about just creating a fat-arrow function?

```js
const withoutVersioning = () => http( { … } );

expect( withoutVersioning ).to.throw( TypeError );
```